llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -9385,6 +9376,16 @@ void VPWidenLoadEVLRecipe::execute(VPTransformState &State) {
   Value *Mask = getMask()
                     ? State.get(getMask(), 0)
                     : Builder.CreateVectorSplat(State.VF, Builder.getTrue());
+  if (isReverse() && getMask()) {
+    VectorType *MaskTy = cast<VectorType>(Mask->getType());
+    Mask = Builder.CreateIntrinsic(
```

Comment:
Would be good to move the code to create the vp.reserves with an all-true mask to a static helper, to be used in various places.

Reply:
Done